File: app/src/main/java/com/bignerdranch/stockwatcher/util/DialogUtils.java
package com.bignerdranch.stockwatcher.util;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;

public class DialogUtils {

    private static final String TAG_DIALOG_PROGRESS = "dialog_progress";

    static void showProgressDialog(FragmentManager fragmentManager,
                                   String message) {
        if (fragmentManager.findFragmentByTag(DialogUtils.TAG_DIALOG_PROGRESS) != null) {
            hideProgressDialog(fragmentManager);
        }
        ProgressDialogFragment dialog = ProgressDialogFragment.newInstance(message);
        dialog.setCancelable(false);
        dialog.show(fragmentManager, TAG_DIALOG_PROGRESS);

    }

    static void hideProgressDialog(FragmentManager fragmentManager) {
        Fragment fragment = fragmentManager.findFragmentByTag(TAG_DIALOG_PROGRESS);
        if (fragment instanceof ProgressDialogFragment) {
            ((ProgressDialogFragment) fragment).dismissAllowingStateLoss();
        }
    }

}

File: app/src/main/java/com/bignerdranch/stockwatcher/util/DialogUtils.kt
package com.bignerdranch.stockwatcher.util

import android.support.v4.app.FragmentManager

private val TAG_DIALOG_PROGRESS = "dialog_progress"

object DialogUtils {

    internal fun showProgressDialog(fragmentManager: FragmentManager, message: String) {
        fragmentManager.findFragmentByTag(TAG_DIALOG_PROGRESS)?.let {
            hideProgressDialog(fragmentManager)
        }
        ProgressDialogFragment.newInstance(message).apply {
            isCancelable = false
            show(fragmentManager, TAG_DIALOG_PROGRESS)
        }
    }

    internal fun hideProgressDialog(fragmentManager: FragmentManager) {
        val fragment = fragmentManager.findFragmentByTag(TAG_DIALOG_PROGRESS)
        if (fragment is ProgressDialogFragment) {
            fragment.dismissAllowingStateLoss()
        }
    }

}
